### PR TITLE
[Inductor XPU] Fix wrong device check before skip concat linear.

### LINKED ISSUE #140917: [Break XPU][Inductor] Wrong device assignment in `addmm_patterns_init()`
### 🐛 Describe the bug

The `device`  in function [`addmm_patterns_init()`](https://github.com/pytorch/pytorch/blob/625c24a7f98a645b6f8758a01d7163a842582ce0/torch/_inductor/fx_passes/freezing_patterns.py#L116C1-L123C19) is assigned with `cuda` or `cpu`,  which missed `xpu`, and lead the actual `device = xpu`  to `device = cpu`.
And after the PR #139048 introduced `config.cpp.enable_concat_linear`, the `addmm_patterns_init()` not works for `xpu`.

### Versions

PyTorch version: 2.6.0a0+git8aaf194
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.29.4
Libc version: glibc-2.35

Python version: 3.9.20 | packaged by conda-forge | (main, Sep 30 2024, 17:49:10)  [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-5.15.47+prerelease23.5.18-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

cc @gujinghui @EikanWang @fengyuan14 @guangyey

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141470
* __->__ #140916

Fix #140917 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
The device initialization doesn't really matter here. We ignore devices in the generated pattern matching searches. If you want to skip for cpp, you should update the `check` functions and check device there.

**Comment by etaf** [comment]
> The device initialization doesn't really matter here. We ignore devices in the generated pattern matching searches. If you want to skip for cpp, you should update the `check` functions and check device there.

Thanks @eellison , I've updated the code, please have a review.

**Review by EikanWang — APPROVED** [review]

File: torch/_inductor/fx_passes/freezing_patterns.py [hunk 1]
```diff
@@ -114,17 +115,17 @@ def register_binary_folding_pattern(pattern, extra_check=_return_true):
...
 def addmm_patterns_init():
-    if torch.cuda.is_available():
-        # workaround https://github.com/pytorch/pytorch/issues/97894
-        device = "cuda"
-    else:
-        device = "cpu"
-        if not config.cpp.enable_concat_linear:
-            return
-
+    # workaround https://github.com/pytorch/pytorch/issues/97894
```
Comment (eellison) [inline torch/_inductor/fx_passes/freezing_patterns.py:118]: this workaround comment is not relevant..

**Review by eellison — APPROVED** [review]

File: torch/_inductor/fx_passes/freezing_patterns.py (hunk 1 above)
Comment (etaf) [reply torch/_inductor/fx_passes/freezing_patterns.py]: resolved, thanks.

**Comment by etaf** [comment]
[xpu / linux-jammy-xpu-2025_0-py3.9 / test (default, 3, 4, linux.idc.xpu)](https://hud.pytorch.org/pr/pytorch/pytorch/140916#33441954164) ([gh](https://github.com/pytorch/pytorch/actions/runs/11996502930/job/33441954164))
inductor/test_compiled_optimizers.py::CompiledOptimizerTests::test_adadelta_maximize_xpu
[xpu / linux-jammy-xpu-2025_0-py3.9 / test (default, 4, 4, linux.idc.xpu)](https://hud.pytorch.org/pr/pytorch/pytorch/140916#33441954259) ([gh](https://github.com/pytorch/pytorch/actions/runs/11996502930/job/33441954259))
inductor/test_compiled_optimizers.py::CompiledOptimizerTests::test_adadelta_weight_decay_xpu
The failed job is known issue: #141466

@pytorchbot merge -i

**Pull request closed without merging**